internal/provider/resource_time_rotating.go

```diff
@@ -0,0 +1,710 @@
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"Read time rotating error",
+				"The rotation_rfc3339 that was supplied could not be parsed as RFC3339.\n\n+"+
+					fmt.Sprintf("Original Error: %s", err),
+			)
+			return
+		}
+
+		if now.After(rotationTimestamp) {
+			log.Printf("[INFO] Expiration timestamp (%s) is after current timestamp (%s), removing from state", state.RotationRFC3339.Value, now.Format(time.RFC3339))
+			resp.State.RemoveResource(ctx)
+			return
+		}
+	}
+
+}
+
+func (t timeRotatingResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var plan, state timeRotatingModelV0
+
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	diags = req.Plan.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if state.RotationYears == plan.RotationYears &&
+		state.RotationMonths == plan.RotationMonths &&
+		state.RotationDays == plan.RotationDays &&
+		state.RotationHours == plan.RotationHours &&
+		state.RotationMinutes == plan.RotationMinutes &&
+		state.RotationRFC3339 == plan.RotationRFC3339 {
+		return
+	}
+
+	timestamp, err := time.Parse(time.RFC3339, plan.ID.Value)
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Update time rotating error",
+			"The ID that was supplied could not be parsed as RFC3339.\n\n+"+
+				fmt.Sprintf("Original Error: %s", err),
+		)
+		return
+	}
+
+	state, err = setRotationValues(&plan, timestamp)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Update time rotating error",
+			"The rotation_rfc3339 timestamp that was supplied could not be parsed as RFC3339.\n\n+"+
+				fmt.Sprintf("Original Error: %s", err),
+		)
+		return
+	}
+	diags = resp.State.Set(ctx, state)
+	resp.Diagnostics.Append(diags...)
+}
+
+func (t timeRotatingResource) Delete(ctx context.Context, request resource.DeleteRequest, response *resource.DeleteResponse) {
+
+}
+
+type timeRotatingModelV0 struct {
+	Day             types.Int64  `tfsdk:"day"`
+	RotationDays    types.Int64  `tfsdk:"rotation_days"`
+	RotationHours   types.Int64  `tfsdk:"rotation_hours"`
+	RotationMinutes types.Int64  `tfsdk:"rotation_minutes"`
+	RotationMonths  types.Int64  `tfsdk:"rotation_months"`
+	RotationRFC3339 types.String `tfsdk:"rotation_rfc3339"`
+	RotationYears   types.Int64  `tfsdk:"rotation_years"`
+	Hour            types.Int64  `tfsdk:"hour"`
+	Triggers        types.Map    `tfsdk:"triggers"`
+	Minute          types.Int64  `tfsdk:"minute"`
+	Month           types.Int64  `tfsdk:"month"`
+	RFC3339         types.String `tfsdk:"rfc3339"`
+	Second          types.Int64  `tfsdk:"second"`
+	Unix            types.Int64  `tfsdk:"unix"`
+	Year            types.Int64  `tfsdk:"year"`
+	ID              types.String `tfsdk:"id"`
+}
+
+func setRotationValues(plan *timeRotatingModelV0, timestamp time.Time) (timeRotatingModelV0, error) {
```

Comment:
The `setRotationValues` could mutate `plan *timeRotatingModelV0` and only return error so that the plan supplied could be used when setting the state.